compiler/damlc/tests/src/DA/Test/ScriptService.hs

```diff
@@ -43,17 +44,25 @@ import Test.Tasty
 import Test.Tasty.HUnit
 import Text.Regex.TDFA
 
-lfVersion :: LF.Version
-lfVersion = LF.versionDefault
-
 main :: IO ()
 main =
+    defaultMain $
+        testGroup
+            "Script Service"
+            [ withResourceCps (withScriptService lfVersion) (testScriptService lfVersion)
+            | lfVersion <- map LF.defaultOrLatestStable [minBound @LF.MajorVersion .. maxBound]
+            ]
+
+withScriptService :: LF.Version -> (SS.Handle -> IO ()) -> IO ()
+withScriptService lfVersion action =
   withTempDir $ \dir -> do
     withCurrentDirectory dir $ do
       setEnv "TASTY_NUM_THREADS" "1" True
 
       -- Package DB setup, we only need to do this once so we do it at the beginning.
-      scriptDar <- locateRunfiles $ mainWorkspace </> "daml-script/daml/daml-script.dar"
+      scriptDar <- locateRunfiles $ case LF.versionMajor lfVersion of
+          LF.V1 -> mainWorkspace </> "daml-script" </> "daml" </> "daml-script.dar"
+          LF.V2 -> mainWorkspace </> "daml-script" </> "daml3" </> "daml3-script.dar"
```

Comment:
I tried to debug the V2 failures but couldn't figure out why it's not picking up daml3-script. @samuel-williams-da might know?

Reply:
I thought this was running on my machine but I had actually only  run it with `-p 2.1`, which works! When inverting the order (v2, then v1), then v2 tests succeed and the v1 ones fail with that error. So there must be some side-effect leaking out of the setup phase. I'll investigate.

Reply:
Interestingly, it fails similarly when running v1 then v1.

Reply:
Fixed it! Needed more ~cowbell~ MVars.